crates/runtime/src/accelerated_table/caching.rs

```diff
@@ -668,11 +669,27 @@ impl CacheRefreshHelper {
                     dataset_name,
                     e
                 );
-                let error_stream = RecordBatchStreamAdapter::new(
-                    fallback_schema,
-                    futures::stream::once(async move { Err(e) }),
-                );
-                Box::pin(error_stream)
+
+                match cached_batches {
```

Comment:
we shouldn't do this, if the items are expired/rotten then they don't logically exist and we should return an error.